libs/sync/snapshot_state.go

```diff
@@ -49,7 +48,8 @@ func NewSnapshotState(localFiles []fileset.File) (*SnapshotState, error) {
 	for _, f := range localFiles {
 		// Compute the remote name the file will have in WSFS
 		remoteName := filepath.ToSlash(f.Relative)
-		isNotebook, _, err := notebook.Detect(f.Absolute)
+		isNotebook, err := f.IsNotebook()
+
```

Comment:
If the underlying file does not exist on the local file system, and we're restoring a remote snapshot, then the file should still be included in the snapshot such that a new sync will issue a delete for the notebook.

Reply:
Yes, indeed, that's what's happening right now (confirmed with test coverage)

Reply:
Ah, I understand now; this works because whether the file is a notebook or not is cached in the underlying object. The lines are blurred w.r.t. responsibilities, between what happens in the bundle code, what happens in `fileset`, and what happens here. I don't have a concrete proposal for a different approach, but this feels a bit scattered. Thoughts?

Reply:
I'd argue that the way things now is quite reasonable:
1. within `snapshot_state` scope if `IsNotebook` fails with whatever reason - we can't proceed further
2. within `file` scope we keep an internal value of `isNotebook` but provide public functions to create files of specific types like `NewNotebookFile`
3. within `development state` scope, we know what is the type of the file we work with and thus can instantiate either `NewNotebookFile` or `NewSourceFile`

I guess the confusion comes from knowing that previously we have to always do calls to FS and now that's not the case anymore